open_gpt/models/modeling.py

```diff
@@ -34,21 +38,25 @@ def __init__(
         self._model_name_or_path = model_name_or_path
         self._adapter_name_or_path = adapter_name_or_path
 
+        self._backend = backend
         self._precision = precision
         self._dtype, self._device = auto_dtype_and_device(precision, device)
 
         self._device_map = device_map or get_device_map(self._device)
 
         self._eval_mode = eval_mode
 
+        self._tensor_parallel_size = tensor_parallel_size
+        self._pipeline_parallel_size = pipeline_parallel_size
```

Comment:
All of these hyperparameters should be advanced configurations and would be better to be determined automatically without manual settings.